File: leetcode/506.cpp
// Lambda expression is cool.

class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& nums) {
        int n = nums.size();
        vector<int> ranks(n);
        for (int i = 0; i < n; i++)
            ranks[i] = i;
        
        sort(ranks.begin(), ranks.end(), [&nums](int p1, int p2) {
            return nums[p1] > nums[p2];
        });
        
        vector<string> output(n);
        for (int i = 0; i < n; i++) {
            if (i == 0) {
                output[ranks[i]] = "Gold Medal";
            } else if (i == 1) {
                output[ranks[i]] = "Silver Medal";
            } else if (i == 2) {
                output[ranks[i]] = "Bronze Medal";
            } else {
                output[ranks[i]] = to_string(i + 1);
            }
        }
        return output;
    }
};

File: leetcode/506.rs
use std::cell::{BorrowMutError, Cell, Ref, RefMut, RefCell};
use std::cmp::{max, min, Ordering};
use std::collections::{HashMap, HashSet, VecDeque};
use std::i32;
use std::mem::swap;
use std::rc::Rc;
use std::iter::repeat;

impl Solution {
    pub fn find_relative_ranks(a: Vec<i32>) -> Vec<String> {
        let n = a.len();
        let mut b = Vec::with_capacity(n);
        for i in 0..n {
            b.push((i, a[i]));
        }
        b.sort_by(|(_, ai), (_, aj)| aj.cmp(&ai));

        let mut ss: Vec<String> = repeat(String::new()).take(n).collect();
        for i in 0..n {
            let (aid, _) = b[i];
            ss[aid] = {
                match i {
                    0 => "Gold Medal".to_string(),
                    1 => "Silver Medal".to_string(),
                    2 => "Bronze Medal".to_string(),
                    _ => (i + 1).to_string(),
                }
            };
        }
        return ss;
    }
}
